moto/apigateway/models.py

```diff
@@ -150,24 +153,124 @@ def delete_integration(self, method_type):
 
 
 class Stage(dict):
-    def __init__(self, name=None, deployment_id=None):
+
+
+    def __init__(self, name=None, deployment_id=None, variables={}):
```

Comment:
Generally not a good idea to have mutable things as default arguments. See [here](http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments) for why. It is one of the bigger gotchas in python IMO. It looks like you have a similar thing in a couple other places too.